logicrunner/builtin/foundation/result.go

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright 2019 Insolar Technologies GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package foundation
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/insolar/insolar/insolar"
+)
+
+type Result struct {
+	Error *Error
+	Returns []interface{}
+}
+
+
+func MarshalMethodResult(returns ...interface{}) ([]byte, error) {
+	result, err := insolar.Serialize(Result{Returns: returns})
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't serialize method returns")
+	}
+
+	return result, nil
+}
+
+func MarshalMethodErrorResult(err error) ([]byte, error) {
+	result, err := insolar.Serialize(Result{Error: &Error{S: err.Error()}})
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't serialize method returns")
+	}
+
+	return result, nil
+}
```

Comment:
Please write unit tests for new procedures.

Reply:
done